From fa546f8f9ef137e18a21f621cef7529c4e6a1788 Mon Sep 17 00:00:00 2001
From: Jason Wu <j.wu@xilinx.com>
Date: Thu, 12 Mar 2015 17:32:19 +1000
Subject: [PATCH 222/827] Documentation: Add documentation for ZynqMP R5 rproc bindings

This patch comes from:
  https://github.com/Xilinx/linux-xlnx.git

Signed-off-by: Jason Wu <j.wu@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
(cherry picked from commit 7c2f3ef3761a8b348c3b90916af93ab962ab6c9e)
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../bindings/remoteproc/r5_remoteproc.txt          | 49 ++++++++++++++++++++++
 1 file changed, 49 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt

diff --git a/Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt b/Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt
new file mode 100644
index 0000000..ec87f4c
--- /dev/null
+++ b/Documentation/devicetree/bindings/remoteproc/r5_remoteproc.txt
@@ -0,0 +1,49 @@
+Xilinx ARM Cortex A53-R5 remoteproc driver
+==========================================
+
+ZynqMP family of devices use two Cortex R5 processors to help with various
+low power / real time tasks.
+
+This driver requires specific ZynqMP hardware design.
+
+ZynqMP R5 RemoteProc Device Node:
+=================================
+A zynqmp_r5_remoteproc device node is used to represent a R5 IP instance
+within ZynqMP SoC.
+
+Required properties:
+--------------------
+ - compatible : Should be "xlnx,zynqmp-r5-remoteproc-1.0"
+ - reg : Address and length of the register set for the device. It
+        contains in the same order as described reg-names
+ - reg-names: Contain the register set names. For direct control method,
+              ipi, rpu_base and apb_base must be provided
+ - interrupts : Interrupt mapping for remoteproc IPI
+ - interrupt-parent : Phandle for the interrupt controller
+
+Optional properties:
+--------------------
+ - firmware : Default firmware image name "rproc-rpu-fw" which can be
+              loaded by remoteproc framework. This value can be override
+              by "firmware" module parameter.
+ - bootmem : R5 boot device (valid string, ocm or tcm), default is ocm.
+ - core_conf : R5 core configuration (valid string - split0 or split1 or
+               lock-step), default is lock-step.
+ - method : RPU and IPI control method - direct, smc, hvc, default is smc.
+ - ipi_dest_mask : IPI channel destination mask. This is used to clear
+         ipi SR and let the rpu know the kernel has completed the work.
+         default mask is "0x100"
+
+Example:
+--------
+	zynqmp-r5-remoteproc@0 {
+		compatible = "xlnx,zynqmp-r5-remoteproc";
+		reg = <0x0 0x3ed00000 0x800000>, <0x0 0xff300000 0x100>, <0x0 0xff9a0000 0x400>, <0x0 0xff5e0000 0x400>;
+		reg-names = "vring0", "ipi", "rpu_base", "apb_base";
+		core_conf = "split0";
+		method = "direct";
+		bootmem = "ocm";
+		firmware = "r5_image";
+		interrupt-parent = <&gic>;
+		interrupts = <0 49 4>;
+	} ;
-- 
2.9.3

